File: frontend/src/types/session.ts
/**
 * Session related types
 */

export interface Session {
  session_id: string;
  name: string;
  framework: string;
  workspace_id: string;
  workspace_path: string;
  creator: string;
  created_at: number;
  updated_at: number;
  is_running?: boolean;
}

export interface SessionCreate {
  name: string;
  framework: string;
}

export interface SessionResponse extends Session {}

File: src/mgx_api/schemas/session.py
"""Session/App related schemas."""
from pydantic import BaseModel


class SessionCreate(BaseModel):
    """Request to create a new session/app."""
    name: str
    framework: str = "fastapi-vite"  # nextjs | fastapi-vite


class SessionResponse(BaseModel):
    """Session/App response."""
    session_id: str
    name: str
    framework: str
    workspace_id: str
    workspace_path: str
    created_at: float
    is_running: bool = False
